tests/test_utils.py

```diff
@@ -137,8 +137,20 @@ def test_merge_consecutive_messages():
     ]
     result = merge_consecutive_messages(messages)
```

Comment:
maybe have a provider keyword so it doesn't mess up openai stuff

Reply:
I reckon the previous implementation above (with the \n instead of the extend method for merging messages) was better and didn't break openAI things. I'd be in favour of not needing to pass provider keywords if possible, the more generic the code the better in my opinion. Ideally once created the initial instructor client, the rest of the code should not have references to the underlying model being code

Reply:
> maybe have a provider keyword so it doesn't mess up openai stuff
OpenAI has a similar way of merging messages of the same role, but this function can also be used by Anthropic exclusively (like what current implementation does), do you mean you also want openai to use this function?

Reply:
> I reckon the previous implementation above (with the \n instead of the extend method for merging messages) was better and didn't break openAI things. I'd be in favour of not needing to pass provider keywords if possible, the more generic the code the better in my opinion. Ideally once created the initial instructor client, the rest of the code should not have references to the underlying model being code

For the sake of not messing up with the current code base I can see you point, but I suppose OpenAI clients doesn't need to merge message anyway, and this merge function is mainly used for Anthropic message API which requires alternating roles in each message, and it's a convenient way to introduce image/tool use messages.